backend/routes/calendar-feed.js

```diff
@@ -0,0 +1,347 @@
+            parts.push('FREQ=YEARLY');
+            if (interval > 1) parts.push(`INTERVAL=${interval}`);
+            break;
+
+        default:
+            return null;
+    }
+
+    // Add end date if specified
+    if (task.recurrence_end_date) {
+        parts.push(`UNTIL=${formatICalDate(task.recurrence_end_date)}`);
+    }
+
+    return parts.length > 0 ? parts.join(';') : null;
+}
+
+/**
+ * Generate a single VEVENT component for a task
+ */
+function generateVEvent(task, hostname) {
+    const lines = [];
+
+    lines.push('BEGIN:VEVENT');
+
+    // UID must be globally unique and persistent
+    lines.push(`UID:task-${task.id}@${hostname}`);
+
+    // Use due_date as the event date (all-day event)
+    // Tasks typically don't have specific times, so we use VALUE=DATE
+    const dueDate = formatICalDate(task.due_date);
+    lines.push(`DTSTART;VALUE=DATE:${dueDate}`);
+    lines.push(`DTEND;VALUE=DATE:${dueDate}`);
+
+    // Created and modified timestamps
+    if (task.created_at) {
+        lines.push(`CREATED:${formatICalDateTime(task.created_at)}`);
+    }
+    if (task.updated_at) {
+        lines.push(`DTSTAMP:${formatICalDateTime(task.updated_at)}`);
+        lines.push(`LAST-MODIFIED:${formatICalDateTime(task.updated_at)}`);
+    } else {
+        lines.push(`DTSTAMP:${formatICalDateTime(new Date())}`);
+    }
+
+    // Task name as summary
+    lines.push(`SUMMARY:${escapeICalText(task.name)}`);
+
+    // Build description: Project first, then description, then note
+    // Only include fields that have content
+    const descParts = [];
+
+    // Project name first (if available)
+    if (task.Project && task.Project.name) {
+        descParts.push(`Project: ${task.Project.name}`);
+    }
+
+    // Task description (the main text field in tududi UI)
+    if (task.description && task.description.trim()) {
+        descParts.push(task.description.trim());
+    }
+
+    // Note field (if it exists and has content)
+    if (task.note && task.note.trim()) {
+        descParts.push(task.note.trim());
+    }
+
+    if (descParts.length > 0) {
+        // Join with double newline for visual separation
+        const descriptionText = descParts.join('\n\n');
+        lines.push(`DESCRIPTION:${escapeICalText(descriptionText)}`);
+    }
+
+    // Add recurrence rule if task is recurring
+    const rrule = buildRRule(task);
+    if (rrule) {
+        lines.push(`RRULE:${rrule}`);
+    }
+
+    lines.push('END:VEVENT');
+
+    return lines.join('\r\n');
+}
+
+/**
+ * Generate the complete iCalendar document
+ */
+function generateICalendar(tasks, calendarName, hostname) {
+    const lines = [];
+
+    // Calendar header
+    lines.push('BEGIN:VCALENDAR');
+    lines.push('VERSION:2.0');
+    lines.push('PRODID:-//tududi//Task Calendar//EN');
+    lines.push('CALSCALE:GREGORIAN');
+    lines.push('METHOD:PUBLISH');
+    lines.push(`X-WR-CALNAME:${escapeICalText(calendarName)}`);
+    lines.push('X-WR-TIMEZONE:UTC');
```

Comment:
Honest question: even though I'm in UTC, would this setting respect both my TZ in Tududi and the TZ of my own calendar?
If I have some time, I'll download it so I can test it here.